### PR TITLE
[SDPA-3114] Add print publication feature.

### PR DESCRIPTION
**Pull request by dylankelly**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-3114

### Changed

1.  Adds Print all page at /:publicationname/print-all
1.  Changes RplPublicationDownloadPrint to display both link to print all and print the current page
1.  Exposes get function on $tideMapping service
1.  Add tests

### Screenshots

app-ripple-pr-543 lagoon vicsdp amazee io_demo-publication_print-all
[Demo Publication.pdf](https://github.com/dpc-sdp/ripple/files/3650226/Demo.Publication.pdf)

### CODE REVIEW

File: packages/ripple-nuxt-tide/modules/publication/pages/PrintAll.vue [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+    const includeConfig = require('./../tide.config')
+    const includes = includeConfig.include.publicationPage
+
+    const publicationPagesData = await context.app.$tide.getContentList('publication_page', filters, includes)
+    const publicationPages = []
+    async function addComponents () {
+      for (let i = 0; i < publicationPagesData.length; i++) {
+        const page = publicationPagesData[i]
+        const componentMapping = await context.app.$tideMapping.getMapping(page.field_landing_page_component, 'landingPageComponents')
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/modules/publication/pages/PrintAll.vue]: `$tideMapping.get()` should work, https://github.com/dpc-sdp/ripple/blob/develop/packages/ripple-nuxt-tide/lib/core/middleware.js#L133

Comment (dylankelly) [reply packages/ripple-nuxt-tide/modules/publication/pages/PrintAll.vue]: it doesn't though, its not exposed in the constructor (try consoling out this.$tideMapping)

File: packages/ripple-nuxt-tide/modules/publication/pages/PrintAll.vue [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    return {
+      publication,
+      publicationPages
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+  @import "~@dpc-sdp/ripple-global/style";
```
[the remark this answers is not in the document]
Comment (dylankelly) [reply packages/ripple-nuxt-tide/modules/publication/pages/PrintAll.vue]: fixed

File: packages/ripple-nuxt-tide/modules/publication/pages/PrintAll.vue (hunk 1 above)
Comment (tim-yao) [reply packages/ripple-nuxt-tide/modules/publication/pages/PrintAll.vue]: @dylankelly as discussed, please remove the get in constructor.

**Pull request merged**